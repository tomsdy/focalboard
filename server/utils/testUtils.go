package utils

import "github.com/stretchr/testify/mock"

var Anything = mock.MatchedBy(func(interface{}) bool { return true })
