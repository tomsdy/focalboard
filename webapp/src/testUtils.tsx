// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IntlProvider} from 'react-intl'
import React from 'react'

export const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

export function mockDOM(): void {
    window.focus = jest.fn()
    document.createRange = () => {
        const range = new Range()

        range.getBoundingClientRect = jest.fn()

        range.getClientRects = () => {
            return {
                item: () => null,
                length: 0,
                [Symbol.iterator]: jest.fn(),
            }
        }

        return range
    }
}
