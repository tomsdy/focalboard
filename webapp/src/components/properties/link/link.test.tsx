// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {render} from '@testing-library/react'

import '@testing-library/jest-dom'
import {wrapIntl} from '../../../testUtils'

import Link from './link'

describe('components/properties/link', () => {
    test('returns link properties correctly', () => {
        const {container} = render(wrapIntl((
            <Link
                value={'https://github.com/mattermost/focalboard'}
                onChange={jest.fn()}
                onSave={jest.fn()}
                onCancel={jest.fn()}
                validator={jest.fn(() => {
                    return true
                })}
            />
        )))
        expect(container).toMatchSnapshot()
    })
})
