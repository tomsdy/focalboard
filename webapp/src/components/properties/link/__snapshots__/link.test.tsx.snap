// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/properties/link returns link properties correctly 1`] = `
<div>
  <div
    class="URLProperty octo-propertyvalue"
  >
    <a
      class="link"
      href="https://github.com/mattermost/focalboard"
      rel="noreferrer"
      target="_blank"
    >
      https://github.com/mattermost/focalboard
    </a>
    <button
      aria-label="Edit"
      class="Button IconButton Button_Edit"
      title="Edit"
      type="button"
    >
      <i
        class="CompassIcon icon-pencil-outline EditIcon"
      />
    </button>
    <button
      aria-label="Copy"
      class="Button IconButton Button_Copy"
      title="Copy"
      type="button"
    >
      <i
        class="CompassIcon icon-content-copy content-copy"
      />
    </button>
  </div>
</div>
`;
